packages/api/tests/endpoints/async-operations/test-create-async-operations.js

```diff
@@ -0,0 +1,480 @@
+  const { testKnex } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeAsyncOperationModel = {
+    create: () => {
+      throw new Error('something bad');
+    },
+    exists: () => Promise.resolve(false),
+    delete: () => Promise.resolve(true),
+  };
+
+  const expressRequest = {
+    body: asyncOperation,
+    testContext: {
+      knex: testKnex,
+      asyncOperationModel: fakeAsyncOperationModel,
+    },
+  };
+
+  const response = buildFakeExpressResponse();
+
+  await post(expressRequest, response);
+
+  t.true(response.boom.badImplementation.calledWithMatch('something bad'));
+
+  t.false(await t.context.esAsyncOperationsClient.exists(
+    asyncOperation.id
+  ));
+  t.false(
+    await t.context.asyncOperationPgModel.exists(t.context.testKnex, {
+      id: asyncOperation.id,
+    })
+  );
+});
+
+test('post() does not write to DynamoDB/Elasticsearch if writing to PostgreSQL fails', async (t) => {
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeAsyncOperationPgModel = {
+    create: () => Promise.reject(new Error('something bad')),
+  };
+
+  const fakeAsyncOperationModel = {
+    delete: () => Promise.resolve(),
+    exists: () => Promise.resolve(false),
+  };
+
+  const expressRequest = {
+    body: asyncOperation,
+    testContext: {
+      asyncOperationPgModel: fakeAsyncOperationPgModel,
+      asyncOperationModel: fakeAsyncOperationModel,
+    },
+  };
+
+  const response = buildFakeExpressResponse();
+
+  await post(expressRequest, response);
+
+  t.true(response.boom.badImplementation.calledWithMatch('something bad'));
+
+  t.false(await t.context.esAsyncOperationsClient.exists(
+    asyncOperation.id
+  ));
+  t.false(await t.context.asyncOperationModel.exists({ id: asyncOperation.id }));
+});
+
+test('post() does not write to DynamoDB/PostgreSQL if writing to Elasticsearch fails', async (t) => {
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const fakeEsClient = {
+    index: () => Promise.reject(new Error('something bad')),
+  };
+
+  const fakeAsyncOperationModel = {
+    delete: () => Promise.resolve(),
+    exists: () => Promise.resolve(false),
+    create: () => Promise.resolve(asyncOperation),
+  };
+
+  const fakeAsyncOperationPgModel = {
+    create: () => Promise.resolve(),
+  };
```

Comment:
Same here: would prefer not to mock these unless there's some reason we need to that I'm not grasping

Reply:
[`7341e50` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/7341e50c4cad294e1c53ae68a8dd5bf599f6568d)